File: gcj/gcj2017/qual/a/a.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

void flip(char *S, int idx, int K) {
    for(int j = idx; j < idx + K; ++j) {
        S[j] = (S[j] == '+') ? '-' : '+';
    }
    return;
}

const int MAX = 1010;
void solve() {
    char S[MAX];
    int K;
    scanf("%s %d", S, &K);
    int len = strlen(S);

    int ans = 0;
    for(int i = 0; i < len - K + 1; ++i) {
        if (S[i] == '-') {
            ans += 1;
            flip(S, i, K);
        }
    }

    if (strchr(S, '-') != 0) {
        printf("IMPOSSIBLE\n");
    } else {
        printf("%d\n", ans);
    }
}
        
int main() {
    int T;
    scanf("%d", &T);
    int t = 0;
    for(t = 0; t < T; ++t) {
        printf("Case #%d: ", t + 1);
        solve();
    }
    return 0;
}

File: gcj/gcj2017/qual/a/a.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

def flip(S, idx, K):
    for j in range(idx, idx + K):
        S[j] = '-' if S[j] == '+' else '+'
    return S

def solve():
    S, K = input().split()
    S = list(S)
    K = int(K)

    ans = 0
    for i in range(len(S) - K + 1):
        if S[i] == '-':
            ans += 1
            S = flip(S, i, K)
    if '-' in S:
        print("IMPOSSIBLE")
    else:
        print(ans)
        
def main():
    T = int(input())
    for t in range(T):
        print ("Case #" + str(t+1) + ": ", end="")
        solve()

main()

